### ISSUE TITLE
Discuss: Should we log activity (URLs submitted + basic analysis results) ?

### ISSUE THREAD

**Issue by mooeypoo** [mooeypoo/rtl.works#20]
TLDR; This is something that was suggested to me, and I think there could be really beneficial aspects to it. However, I am worried to make sure I don't go against privacy expectations or good practices.

So, this thread is about two questions:

1. Whether activity information should be collected at all
2. What information is good to collect, considering useful data vs. privacy

Benefits of collecting data:
- Technical knowledge: If I can follow the use cases that are used, I can try and anticipate technical problems as well as the direction that users seem to expect the service to go to. I can see cases where this can be extremely useful for continued development.
- Statistics: This one is the strongest reason yet. If I collect basic usage statistics -- the URL used + very basic information about the analysis done at that date with that site -- I can then maybe use it to produce very useful statistics about websites online. For example, I can publish periodic results for things like:
-- How many analyzed sites have characters in both directionalities
-- How many analyzed sites use 'dir' attribute in the html or body tags
-- Analysis of languages on websites, charsets, etc.

So, like any statistics gathering, this can go from small to big, and the question is how 'big' to go before privacy is an issue.

The information gathered could be basic - URLs + basic results of analysis, but we can also add location information (general) like origin country for the request. It probably isn't a must, but it might be interesting to see over time what countries use the service.

I'd love to get opinions on this before I start collecting information the wrong way. Also, if you think there is more (or less) interesting data we could collect without hurting users' privacy, do share.

**Comment by mattflaschen** [comment]
I think best practice here is:
* Only log information that is actually useful.
* Inform users (right next to the form) in easy-to-read language what is being collected, something like "In order to improve the service and understand RTL usage, we collect and store the URLs submitted, as well as their results.  This info is only published in aggregate".
* Have a more detailed privacy policy separate from the "quick read" version above, with exact details on what is collected, what purpose it is used for, and how long it is stored.

Certain information could also be useful to deal with any abuse of the API (e.g. IP logs of who is using it, possibly short-term).
